### ISSUE TITLE
12$: error on optional param followed by required param

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#854]
The following code is invalid TypeScript:
```typescript
function f(x?, y) {}
```
Report a diagnostic.

**Comment by AlekLefebvre** [comment]
I'd like to work on this issue.

I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task

**Comment by strager** [comment]
@AlekLefebvre #860, claimed by you, is in progress right now. You can claim a new task only after that task is completed.

**Comment by AlekLefebvre** [comment]
Hi @strager ,
Now that #860 is closed, can I work on this issue?
Thanks,

**Comment by strager** [comment]
> can I work on this issue?

Yes
